service/storage_proxy.cc

```diff
@@ -3296,6 +3308,23 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit), std::monostate(), is_cancellable::no);
 }
 
+// After sending the write to replicas(targets), the replicas might generate send view updates. If the number of view updates
+// that the replica is already sending is too high, the replica should not send any more view updates, and consequently,
+// we should not send the write to the replica.
+template<typename Range>
+bool storage_proxy::can_update_views(const Range& targets, const schema_ptr& s, db::write_type type) const {
+    static constexpr double mv_admission_control_threshold = 0.8;
```

Comment:
It's a fraction of that other threshold (which is also just the view backlog limit). It could be useful to make both of them configurable (maybe for some tests), but I'm not sure we can actually select "optimal" values for them and make them useful in practice.
Note that as @piodul mentioned in https://github.com/scylladb/scylladb/pull/18334#discussion_r1695412481, using 0.8 here effectively lowers the max mv flow control delay to 512ms (if we were to use a longer delay we will now reject the write). If we want to make this configurable we might want to consider making that delay configurable first (I guess we have https://github.com/scylladb/scylladb/issues/18187 for that)

Reply:
Right, https://github.com/scylladb/scylladb/issues/18187 says that 1s has this special magical meaning that we need to reconsider, and this patch changes it to 0.5 seconds for no obvious reason. Maybe the code here should use 1.0 and not 0.8, and then to avoid the per-replica code accidentally kicking in before this one does, change the >1 in that code to >1.1 or something?

Reply:
Ok, I tried to achieve something of the sort in the rebase: now we reach the 1s delay on the coordinator when we approach the admission control limit. At the same time we still reject replica writes only when we use all `max_memory_pending_view_updates()` bytes for the backlog